### DISCUSSION TITLE
[supabase_flutter]How to Query from multiple columns using logical AND or OR

### DISCUSSION THREAD

**Discussion by maheshj01** [supabase/supabase#15989] [Questions] [answered]
I have a database schema 

```SQL
word_table(id, word, meaning, synonyms, examples,created_at, updated_at)
```

I am trying to implement a search feature on the client side by trying to query from this table based on word,meaning, and synonyms (word and meaning are of type text, and synonyms and examples has a type Array)

This is the SQL query I am trying to create using Supabase Flutter

```SQL
SELECT *
FROM word_table
WHERE word ILIKE '%deception%'
   OR 'deception' ILIKE ANY (synonyms)
   OR meaning ILIKE '%deception%';
```

I did try something like this with no success

```dart
 final response = await _supabase
        .from(tableName)
        .select()
         .ilike('word', "%${columnValue}%")
        .or(
          'meaning.ilike.%$columnValue%',
        )
        .execute();
```        
        
cc: @dshukertjr

**Comment by dshukertjr** [comment, marked as answer]
You should be able to do the following:

```dart
final response = await _supabase
        .from(tableName)
        .select()
        .or('word.ilike.%$columnValue%,meaning.ilike.%$columnValue%');
```

https://supabase.com/docs/reference/dart/or

Note that if you are trying to compose a complex queries, it is probably easier to just create an database function and call it via the rpc method.
